tests/unit/nupic/research/flat_spatial_pooler_compatability_test.py

```diff
@@ -1,533 +0,0 @@
-#! /usr/bin/env python
-# ----------------------------------------------------------------------
-# Numenta Platform for Intelligent Computing (NuPIC)
-# Copyright (C) 2013, Numenta, Inc.  Unless you have an agreement
-# with Numenta, Inc., for a separate license for this software code, the
-# following terms and conditions apply:
-#
-# This program is free software: you can redistribute it and/or modify
-# it under the terms of the GNU General Public License version 3 as
-# published by the Free Software Foundation.
-#
-# This program is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
-# See the GNU General Public License for more details.
-#
-# You should have received a copy of the GNU General Public License
-# along with this program.  If not, see http://www.gnu.org/licenses.
-#
-# http://numenta.org/licenses/
-# ----------------------------------------------------------------------
-
-import cPickle as pickle
-import numpy
-import unittest2 as unittest
-
-from nupic.support.unittesthelpers.algorithm_test_helpers import (
-  getNumpyRandomGenerator, convertPermanences )
-
-from nupic.bindings.algorithms import FlatSpatialPooler as CPPFlatSpatialPooler
-from nupic.bindings.math import GetNTAReal, Random
-from nupic.research.flat_spatial_pooler import (
-  FlatSpatialPooler as PyFlatSpatialPooler)
-
-realType = GetNTAReal()
-uintType = "uint32"
-NUM_RECORDS = 100
-
-
-
-class FlatSpatialPoolerCompatabilityTest(unittest.TestCase):
-
-  def setUp(self):
-    # Set to 1 for more verbose debugging output
-    self.verbosity = 0
-    
-
-  def assertListAlmostEqual(self, alist, blist):
-    self.assertEqual(len(alist), len(blist),
-                     "Lists have different length")
-    for idx, val in enumerate(alist):
-      self.assertAlmostEqual(
-        val,
-        blist[idx],
-        places = 3,
-        msg = "Lists different at index %d with values %g and %g"
-              % (idx, val, blist[idx]))
-
-
-  def compare(self, pySp, cppSp):
-    self.assertAlmostEqual(pySp.getNumColumns(),
-                           cppSp.getNumColumns())
```

Comment:
I think this c++/py compatibility test should stay! (ofc ported to the new SP implementation, not FlatSP)